src/reactor/mod.rs

```diff
@@ -544,18 +547,18 @@ impl Inner {
     fn cancel_timeout(&mut self, token: usize) {
         debug!("cancel a timeout: {}", token);
         let pair = self.timeouts.remove(token);
-        if let Some((Some(slot), _state)) = pair {
+        if let (Some(slot), _state) = pair {
```

Comment:
If `cancel_timeout` could be called twice for the same token in the past, this change is invalid. 0.3.0 would return `None` if the token did not exist; 0.4.0 will panic. I haven't vetted if `cancel_timeout` can be called twice.